python/pylibcugraph/pylibcugraph/_cugraph_c/graph.pxd

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from pylibcugraph._cugraph_c.cugraph_api cimport (
+    bool_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_host_array_view_t,
+)
+
+
+cdef extern from "cugraph_c/graph.h":
+
+    ctypedef struct cugraph_graph_t:
+        pass
+
+    ctypedef struct cugraph_graph_properties_t:
+        bool_t is_symmetric
+        bool_t is_multigraph
+
+    cdef cugraph_error_code_t \
+         cugraph_sg_graph_create(
+             const cugraph_resource_handle_t* handle,
+             const cugraph_graph_properties_t* properties,
+             const cugraph_type_erased_device_array_view_t* src,
+             const cugraph_type_erased_device_array_view_t* dst,
+             const cugraph_type_erased_device_array_view_t* weights,
+             bool_t store_transposed,
+             bool_t renumber,
+             bool_t check,
+             cugraph_graph_t** graph,
+             cugraph_error_t** error)
+
+    # This may get renamed to cugraph_graph_free()
+    cdef void \
+        cugraph_sg_graph_free(
+            cugraph_graph_t* graph
+        )
+
+    cdef cugraph_error_code_t \
+        cugraph_mg_graph_create(
+            const cugraph_resource_handle_t* handle,
+            const cugraph_graph_properties_t* properties,
+            const cugraph_type_erased_device_array_view_t* src,
+            const cugraph_type_erased_device_array_view_t* dst,
+            const cugraph_type_erased_device_array_view_t* weights,
+            const cugraph_type_erased_host_array_view_t* vertex_partition_offsets,
+            const cugraph_type_erased_host_array_view_t* segment_offsets,
```

Comment:
Better remove `vertex_partition_offsets` and `segment_offsets`. These are unnecessary if we use `create_graph_from_edgelist` and this involves too much implementation details.

Reply:
Suggest a FIXME for this.  The MG version will be worked on next release and we'll figure out what changes are required here.

Reply:
added a FIXME in the next commit.